docs/reference/developer/adr/0012-rebin-should-apply-irreducible-masks.rst

```diff
@@ -0,0 +1,51 @@
+ADR 0012: Rebin should apply irreducible masks
```

Comment:
Talked to @arm61. He said that applying the masks as in this PR is not desirable. He doesn't have a concrete use case because he can usually normalise before rebinning. Suggested we talk to @g5t because he uses rebinning a lot. But generally, losing the information that a bin was masked could be a problem.
Also, rebinning is not in the same category as summing because it preserves the dimension.

As an alternative, we could have an option to either apply or preserve masks. Though that would make the implementation more complicated. So maybe providing a function to easily apply irreducible masks would help. So users can do it if they don't want to propagate masks.

Reply:
This would imply that we also need to change the behavior in `bin`?

Reply:
Furthermore, we would also need to change the behavior of `sum` and all reduction operations, since those also drop information about presence of masks.

Reply:
> Furthermore, we would also need to change the behavior of sum and all reduction operations, since those also drop information about presence of masks.

Those operations completely remove the masked dims. I don't think you can propagate a meaningful mask. Except maybe multi-dimensional masks. How are they handled?

Reply:
> I don't think you can propagate a meaningful mask.

You can mask an output element if any of the input elements contributing to it is masked. I agree that it is not *useful* (which is why it isn't doing that), but I would think it is *meaningful*. Then my question is: Why would we treat things differently in `rebin`, what is the argument for that, except for "in most cases summation combines many input elements whereas rebin combines only a few" (which is in my eyes not a valid argument).

Reply:
Currently, if you rebin a dimension into a single bin, as soon as one element along the dimension is masked, you will have the single output bin masked. This is then different from applying the sum along that dimension, where only the single masked element is ignored.

That said, maybe `sum` and `rebin` don't need to do the same thing, because they are two different operations after all?

Reply:
> That said, maybe sum and rebin don't need to do the same thing, because they are two different operations after all?

Can you elaborate? Which part of the argument above (regarding needing to know if there were masked bins) applies to `rebin` but not to `sum`?

Reply:
I guess I was just trying to say that `sum` and `rebin` are doing slightly different things, and this comes from the fact that `rebin` will split old bins into new bins. While `sum` will only work on a range of full bins.

But I do agree that if you are rebinning to a single bin, you should do the same as a `sum`.